### PR TITLE
RQ-789: `RQNetworkTable` integration in session recording

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## 📜 Summary of changes:

https://github.com/requestly/requestly/assets/61556757/3b18d9dc-3c98-42ce-98f6-ef5794cd9cfe

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/lib/design-system/components/RQNetworkTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export const RQNetworkTable: React.FC<RQNetworkTableProps> = ({ logs, sessionRecordingStartTime = 0 }) => {
+  return (
+    <div className="rq-network-table-container">
+      <GenericNetworkTable
+        logs={logs}
+        isFailed={(log: RQNetworkLog) => {
+          const { status } = log.entry.response;
+          return !status || (status >= 400 && status <= 599);
+        }}
```
Comment (wrongsahil) [inline app/src/lib/design-system/components/RQNetworkTable/index.tsx:26]: This logic can be moved in `GenericNetworkTable`. Doesn't seem to be related to Requestly

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: app/src/lib/design-system/components/RQNetworkTable/index.tsx (hunk 1 above)
Comment (rohanmathur91) [reply app/src/lib/design-system/components/RQNetworkTable/index.tsx]: What if others have different criteria for a failed request?

**Review by wrongsahil — APPROVED** [review]
